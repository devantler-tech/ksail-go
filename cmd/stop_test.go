package cmd_test

import (
	"testing"

	"github.com/devantler-tech/ksail-go/cmd"
	"github.com/devantler-tech/ksail-go/cmd/internal/testutils"
)

func TestNewStopCmd(t *testing.T) {
	t.Parallel()

	testutils.TestSimpleCommandCreation(t, testutils.SimpleCommandTestData{
		CommandName:   "stop",
		NewCommand:    cmd.NewStopCmd,
		ExpectedUse:   "stop",
		ExpectedShort: "Stop the Kubernetes cluster",
	})
}

func TestStopCmd_Execute(t *testing.T) {
	t.Parallel()

	testutils.TestSimpleCommandExecution(t, testutils.SimpleCommandTestData{
		CommandName: "stop",
		NewCommand:  cmd.NewStopCmd,
	})
}

func TestStopCmd_Help(t *testing.T) {
	t.Parallel()

	testutils.TestSimpleCommandHelp(t, testutils.SimpleCommandTestData{
		CommandName: "stop",
		NewCommand:  cmd.NewStopCmd,
	})
}
