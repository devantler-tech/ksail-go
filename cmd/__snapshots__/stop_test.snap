[TestStopCmdHelp - 1]
Stop the Kubernetes cluster without removing it.

Usage:
  stop [flags]

Flags:
  -c, --context string              Kubernetes context of cluster to stop (default "kind-ksail-default")
  -d, --distribution Distribution   Kubernetes distribution to stop (default Kind)
  -h, --help                        help for stop

---

[TestStopCmdExecute - 1]
✔ Cluster stopped successfully (stub implementation)
► Distribution: Kind
► Context: kind-ksail-default

---